crypto/fipsmodule/ecdh/ecdh.c

```diff
@@ -131,6 +130,15 @@ int ECDH_compute_shared_secret(uint8_t *buf, size_t *buflen, const EC_POINT *pub
   return ret;
 }
 
+int ECDH_compute_shared_secret(uint8_t *buf, size_t *buflen,
+                               const EC_POINT *pub_key,
+                               const EC_KEY *priv_key) {
+  boringssl_ensure_ecc_self_test();
+
+  return ecdh_compute_shared_secret_no_self_test(buf, buflen, pub_key,
```

Comment:
OK, but we could prevent the same thread from entering `run_self_test_ecc` twice I guess. And we wouldn't need two functions?

Reply:
I'll keep this for now, so we're following our original pattern.
I'm curious though, is there a way to do that? Seems like `CRYPTO_once` only blocks concurrent threads from accessing the original executing thread, is there something that would help us block the original thread execution as well

Reply:
yeah, just a thread-local semaphore